.scripts/build_steps.sh

```diff
@@ -38,6 +38,7 @@ pushd /home/conda/staged-recipes/recipes > /dev/null
 if [ "${AZURE}" == "True" ]; then
     git fetch --force origin main:main
 fi
+git config --global --add safe.directory /home/conda/staged-recipes
```

Comment:
Why is this needed?

Reply:
```
+ echo 'Finding recipes merged in main and removing them from the build.'
Finding recipes merged in main and removing them from the build.
+ pushd /home/conda/staged-recipes/recipes
+ '[' False == True ']'
+ git ls-tree --name-only main -- .
+ xargs -I '{}' sh -c 'rm -rf ~/staged-recipes-copy/recipes/{} && echo Removing recipe: {}'
fatal: detected dubious ownership in repository at '/home/conda/staged-recipes'
To add an exception for this directory, call:

	git config --global --add safe.directory /home/conda/staged-recipes
Traceback (most recent call last):
  File "/home/dching/Documents/staged-recipes/build-locally.py", line 101, in <module>
    main()
  File "/home/dching/Documents/staged-recipes/build-locally.py", line 95, in main
    run_docker_build(ns)
  File "/home/dching/Documents/staged-recipes/build-locally.py", line 33, in run_docker_build
    subprocess.check_call([script])
  File "/usr/lib64/python3.11/subprocess.py", line 413, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['.scripts/run_docker_build.sh']' returned non-zero exit status 1.
```
I get the following error message when using the `build-locally.py`

Reply:
Thanks for the clarification 🙏

Could we please split this out as a separate issue or PR?

It seems unrelated to the changes here (and we may want to evaluate our options for solving it including whether the Docker images themselves need changes)

Reply:
Yes

Reply:
Sounds good. Please feel free to link an issue here (once opened). Agree we should figure out some kind of solution. Think it may even be needed in feedstocks

Reply:
How did you guys get around to this issue I am hitting the same thing when using the build_locally.py script

Reply:
Add the topic line of this conversation to the build_steps script.

Reply:
What I wonder is that this change cannot make it to the feedstock project, thus I would have to add it anytime I want to build locally, would there be an alternative?

Reply:
If the feedstock hasn't been updated in a while, please make sure to [re-render]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#dev-update-rerender )